content/en/docs/reference/node/kubelet-files.md

```diff
@@ -0,0 +1,142 @@
+State files are located in the kubelet's base directory, also termed the _root directory_
+(but not the same as `/`, the node root directory). You can configure the base directory
+for the kubelet
+using the kubelet command line argument `--root-dir`.
+
+Names of files:
+
+- `memory_manager_state` for the [Memory Manager](/docs/tasks/administer-cluster/memory-manager/)
+- `cpu_manager_state` for the [CPU Manager](/docs/tasks/administer-cluster/cpu-management-policies/)
+- `kubelet_internal_checkpoint` for [Device Manager](/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#device-plugin-integration-with-the-topology-manager)
+- `dra_manager_state` for [DRA](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/)
+
+### Pod status checkpoint storage {#pod-status-manager-state}
+
+{{< feature-state feature_gate_name="InPlacePodVerticalScaling" >}}
+
+If your cluster has  
+[in-place Pod vertical scaling](/docs/concepts/workloads/autoscaling/#in-place-resizing)  
+enabled ([feature gate](/docs/reference/command-line-tools-reference/feature-gates/)  
+name `InPlacePodVerticalScaling`), then the kubelet stores a local record of Pod status.  
+
+The file name is `pod_status_manager_state` within the kubelet base directory
+(`/var/lib/kubelet` by default on Linux; configurable using `--root-dir`).
+
+### Container runtime
+
+Kubelet communicates with the container runtime using socket configured via the
+configuration parameters:
+
+- `containerRuntimeEndpoint` for runtime operations
+- `imageServiceEndpoint` for image management operations
+
+The actual values of those endpoints depend on the container runtime being used.
+
+### Device plugins
+
+The kubelet exposes a socket at the path `/var/lib/kubelet/device-plugins/kubelet.sock` for
+various [Device Plugins to register](/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#device-plugin-implementation).
+
+When a device plugin registers itself, it provides its socket path for the kubelet to connect.
+
+The device plugin socket should be in the directory `device-plugins` within the kubelet base
+directory. On a typical Linux node, this means `/var/lib/kubelet/device-plugins`.
+
+### Pod resources API
+
+[Pod Resources API](/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#monitoring-device-plugin-resources)
+will be exposed at the path `/var/lib/kubelet/pod-resources`.
+
+### DRA and CSI plugins
+
+The kubelet looks for socket files created by device plugins managed via [DRA](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/)
+or storage plugins, and then attempts to connect
+to these sockets. The directory that the kubelet looks in is `plugins_registry` within the kubelet base
+directory, so on a typical Linux node this means `/var/lib/kubelet/plugins_registry`.
+
+The types of plugins that can place socket files into that directory are:
+
+- CSI plugins
+- DRA plugins
```

Comment:
+ Device plugins can also register through the same mechanism: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1575-L1582

Reply:
wow, I didn't know. Thanks!

Reply:
Looks resolved